.github/workflows/packages-build-manager.yml

```diff
@@ -0,0 +1,158 @@
+        required: false
+        default: 'auto'
+        type: string
+      architecture:
+        type: string
+        required: true
+      system:
+        type: string
+        required: true
+      os_name:
+        default: ""
+        type: string
+        required: false
+      revision:
+        default: "0"
+        type: string
+        required: false
+      is_stage:
+        type: boolean
+        required: false
+      debug:
+        type: boolean
+        required: false
+      checksum:
+        type: boolean
+        required: false
+
+jobs:
+  Build-manager-packages:
+    if: |
+    runs-on: ubuntu-latest
+    name: Build ${{ inputs.system }} wazuh-manager on ${{ inputs.architecture }}
+
+    steps:
+      - name: Cancel previous runs
+        uses: fkirc/skip-duplicate-actions@master
+        with:
+          cancel_others: 'true'
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          skip_after_successful_duplicate: 'false'
+
+      - uses: actions/checkout@v4
+
+      - name: Set tag and container name
+        run: |
+          VERSION=$(sed 's/^v\([0-9]*\.[0-9]*\.[0-9]*\)/\1/' $GITHUB_WORKSPACE/src/VERSION)
+          if [ "${{ inputs.docker_image_tag }}" == "auto" ]; then echo "TAG=$VERSION" >> $GITHUB_ENV;
+          elif [ "${{ inputs.docker_image_tag }}" == "developer" ]; then echo "TAG=${{ github.ref_name }}" >> $GITHUB_ENV;
+          else echo "TAG=${{ inputs.docker_image_tag }}" >> $GITHUB_ENV; fi
+          if [ ${{ inputs.system }} == 'rpm' ]; then 
+            if [ ${{ inputs.architecture }} == 'x86_64' ]; then arch=amd64;
+            else arch=arm64; fi
+          else arch=${{ inputs.architecture }}; fi
+          echo "ARCH=$arch" >> $GITHUB_ENV;
+          echo "CONTAINER_NAME=pkg_${{ inputs.system }}_manager_builder_${arch}" >> $GITHUB_ENV
+
+      - name: Download docker image for package building
+        run: |
+          bash $GITHUB_WORKSPACE/.github/actions/ghcr-pull-and-push/pull_image_from_ghcr.sh ${{ secrets.GITHUB_TOKEN }} ${{ github.actor}} $CONTAINER_NAME ${{ env.TAG }}
+
+      - name: Build ${{ inputs.system }} wazuh-manager on ${{ inputs.architecture }}
+        working-directory: packages
+        run: |
+          REVISION=${{ inputs.revision }}
+          FLAGS="-t manager --dont-build-docker -j $(nproc) "
+          if [ -z "$REVISION" ]; then FLAGS+="-r 0 "; else FLAGS+="-r $REVISION "; fi
+          if [ "${{ inputs.is_stage }}" == "true" ]; then FLAGS+="--is_stage "; fi
+          if [ "${{ inputs.checksum }}" == "true" ] || [ ${{ inputs.is_stage}} == "true" ]; then FLAGS+="--checksum "; fi
+          if [ "${{ inputs.debug}}" == "true" ]; then FLAGS+="--debug "; fi
+          echo $FLAGS
+          bash generate_package.sh -a ${{ env.ARCH }} --tag ${{ env.TAG }} --system ${{ inputs.system }} $FLAGS
+
+      - name: Test install built manager
+        run: |
+          if [ ${{ inputs.system }} == 'rpm' ]; then TESTING_DOCKER_IMAGE=centos:latest;
```

Comment:
Did you try using the builder container for this? It could avoid downloading a new container.
This works for me:
`$CONTAINER_NAME:${{ env.TAG }}`